fgax/options.go

```diff
@@ -0,0 +1,131 @@
+package fgax
+
+import (
+    openfga "github.com/openfga/go-sdk"
+    ofgaclient "github.com/openfga/go-sdk/client"
+)
+
+const (
+    // HeaderXRequestID is the header name for request id
+    HeaderXRequestID = "X-Request-Id"
+)
+
+// RequestOptions holds per-request options that can be applied to OpenFGA requests
+type RequestOptions struct {
+    Headers     map[string]string
+    Consistency *openfga.ConsistencyPreference
+}
+
+// RequestOption is a functional option for RequestOptions
+type RequestOption func(*RequestOptions)
+
+// WithRequestIDHeader sets the X-Request-Id header for the request
+func WithRequestIDHeader(id string) RequestOption {
+    return func(ro *RequestOptions) {
+        if ro.Headers == nil {
+            ro.Headers = map[string]string{}
+        }
+
+        ro.Headers[HeaderXRequestID] = id
+    }
+}
+
+// WithCustomHeader sets a custom header for the request
+func WithCustomHeader(header, value string) RequestOption {
+    return func(ro *RequestOptions) {
+        if ro.Headers == nil {
+            ro.Headers = map[string]string{}
+        }
+
+        ro.Headers[header] = value
+    }
+}
+
+// WithHighConsistency sets the consistency preference to higher consistency
+func WithHighConsistency() RequestOption {
+    return func(ro *RequestOptions) {
+        c := openfga.CONSISTENCYPREFERENCE_HIGHER_CONSISTENCY
+        ro.Consistency = &c
+    }
+}
+
+// getRequestOptions aggregates functional RequestOptions into a RequestOptions struct
+func getRequestOptions(opts ...RequestOption) RequestOptions {
+    ro := RequestOptions{}
```

Comment:
Instead of each `getXOptions` setting this, we should be able to default the consistency here, and leave it out  on the other function, e.g. 

Adding here 
```suggestion
    ro := RequestOptions{
        Consistency: defaultConsistency,
    }
```

`getBatchCheckOptions`, `getReadOptions`, and `getCheckOptions` don't need to set it in the default setup, e.g. 

can just be: 
```
    o := ofgaclient.BatchCheckOptions{
        MaxBatchSize:        &batchSizeLimit,
        MaxParallelRequests: &batchParallelLimit,
    }
```